### DISCUSSION TITLE
Heads up for some upcoming changes

### DISCUSSION THREAD

**Discussion by gkdn** [google/j2cl#258] [Announcements]
I wanted to give a heads up some upcoming multiple changes that may disrupt your work if you are following the head version closely.

- [x] Will change minimum supported version of Bazel to 7+.
- [x] Will start working towards [Bzlmod support](https://bazel.build/external/migration) which make things smoother for Bazel going forward.
- [x] Will start adding language features towards Java 21 (current supported language level is Java 11).
- [x] With Java 21 transition, J2CL will switch to javac as its frontend and JDT will be dropped.

**Comment by treblereel** [comment]
@gkdn 
Thank you so much for the work you do! What are the disadvantages/potential pitfalls we might face when switching from JDT to Javac? How can we prepare for this in advance?

**Reply by gkdn** [reply]
Most common difference we observed between the frontends was where some edge cases they inferred different types. However since J2CL code is almost always also compiled with Javac (either bytecode for J2CL classpath or cross-compilation with other platforms), I don't anticipate you to observe that difference (if you hit that, you probably already needed to work around it).

So I'm hoping that it would be a seamless transition for end users.
